.cargo/config

```diff
@@ -1,6 +1,12 @@
 [alias]
 xtask = "run --package xtask --"
 
+# Faster linking in CI for doctests where lld is available
+[target.'cfg(any(target_os = "windows", target_os = "unix"))']
```

Comment:
I don't think `target_os = "unix"` is valid?

Reply:
Indeed, https://doc.rust-lang.org/reference/conditional-compilation.html#target_os, so either `target_os = "linux"` or `target_familiy = "unix"`.

I think using the former is safer and would suffice for our CI, but considering https://github.com/PyO3/pyo3/pull/2836#discussion_r1057623555, we might want to handle this via the workflow configuration in any case?